File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsinvoicepagedresponsemodel.go
package shared

// CodatDataContractsDatasetsInvoicePagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type CodatDataContractsDatasetsInvoicePagedResponseModel struct {
	Links        *CodatDataContractsDatasetsInvoicePagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                                      `json:"pageNumber,omitempty"`
	PageSize     *int                                                      `json:"pageSize,omitempty"`
	Results      []CodatDataContractsDatasetsInvoice                       `json:"results,omitempty"`
	TotalResults *int                                                      `json:"totalResults,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsinvoicepagedresponsemodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsInvoice } from "./codatdatacontractsdatasetsinvoice";
import { CodatDataContractsDatasetsInvoicePagedResponseLinksModel } from "./codatdatacontractsdatasetsinvoicepagedresponselinksmodel";
import { Expose, Type } from "class-transformer";


// CodatDataContractsDatasetsInvoicePagedResponseModel
/** 
 * Used to represent what can be returned by an endpoint that supports paging.
 * Usable with the [ProducesResponseType] attribute on a controller action.
**/
export class CodatDataContractsDatasetsInvoicePagedResponseModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "_links" })
  @Type(() => CodatDataContractsDatasetsInvoicePagedResponseLinksModel)
  links?: CodatDataContractsDatasetsInvoicePagedResponseLinksModel;

  @SpeakeasyMetadata()
  @Expose({ name: "pageNumber" })
  pageNumber?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ elemType: CodatDataContractsDatasetsInvoice })
  @Expose({ name: "results" })
  @Type(() => CodatDataContractsDatasetsInvoice)
  results?: CodatDataContractsDatasetsInvoice[];

  @SpeakeasyMetadata()
  @Expose({ name: "totalResults" })
  totalResults?: number;
}